cmd/eksctl/main.go

```diff
@@ -18,67 +18,67 @@ import (
 	"github.com/weaveworks/eksctl/pkg/ctl/utils"
 )
 
-var rootCmd = &cobra.Command{
-	Use:   "eksctl [command]",
-	Short: "a CLI for Amazon EKS",
-	Run: func(c *cobra.Command, _ []string) {
-		if err := c.Help(); err != nil {
-			logger.Debug("ignoring error %q", err.Error())
-		}
-	},
+func addCommands(rootCmd *cobra.Command, flagGrouping *cmdutils.FlagGrouping) {
+	rootCmd.AddCommand(versionCmd(flagGrouping))
+	rootCmd.AddCommand(create.Command(flagGrouping))
+	rootCmd.AddCommand(delete.Command(flagGrouping))
+	rootCmd.AddCommand(get.Command(flagGrouping))
+	rootCmd.AddCommand(update.Command(flagGrouping))
+	rootCmd.AddCommand(scale.Command(flagGrouping))
+	rootCmd.AddCommand(drain.Command(flagGrouping))
+	rootCmd.AddCommand(utils.Command(flagGrouping))
+	rootCmd.AddCommand(completion.Command(rootCmd))
 }
 
-func init() {
+func main() {
+	cobra.EnableCommandSorting = false
```

Comment:
I think we should note this and mention in the release notes that the command list would now be in the order of importance:

```Commands:
  eksctl version          Output the version of eksctl
  eksctl create           Create resource(s)
  eksctl delete           Delete resource(s)
  eksctl get              Get resource(s)
  eksctl update           Update resource(s)
  eksctl scale            Scale resources(s)
  eksctl drain            Drain resource(s)
  eksctl utils            Various utils
  eksctl completion       Generates shell completion scripts
  eksctl help             Help about any command
```

Also, what do you think about moving `eksctl version` below in the list? `kubectl` for e.g., lists version at the end.

Reply:
What do you mean in order of importance? The version is the most important command?

Reply:
`version` is not important, which is why I suggested moving it to the end.

Reply:
What I was mostly trying to address is disable alphabetical sorting and make it sort in order of declaration. But oder of declaration should be the oder importance, so I'll move version command to the bottom.

Reply:
New look:
```
The official CLI for Amazon EKS

Usage: eksctl [command] [flags]

Commands:
  eksctl create           Create resource(s)
  eksctl get              Get resource(s)
  eksctl update           Update resource(s)
  eksctl delete           Delete resource(s)
  eksctl scale            Scale resources(s)
  eksctl drain            Drain resource(s)
  eksctl utils            Various utils
  eksctl completion       Generates shell completion scripts
  eksctl version          Output the version of eksctl
  eksctl help             Help about any command

Common flags:
  -C, --color string   toggle colorized logs (valid options: true, false, fabulous) (default "true")
  -h, --help           help for this command
  -v, --verbose int    set log level, use 0 to silence, 4 for debugging and 5 for debugging with AWS debug logging (default 3)

Use 'eksctl [command] --help' for more information about a command.
```

Reply:
Looks good!